docs/RationaleSeq.md

```diff
@@ -106,3 +106,48 @@ information.
 - Initial value: this register is uninitialized. Using an uninitialized value
 results in undefined behavior. We will add an `initialValue` attribute if
 this proves insufficient.
+
+## The Register Operation
+
+`seq.reg` abstractly models a generic memory element that can be implemented
+as various physical memory elements. This attempts to capture most behaviours
+expressable by a `reg` in SystemVerilog.
+
+This is an abstract operation that should be further lowered down to either
+vendor-specific primtives or to the SV dialect.
+
+The register operation contains the following operands:
+
+- clock and its edge sensitivity: posedge, negedge, edge-triggered, or
+  unclocked. An unclocked register can be used to express a latch.
```

Comment:
Does it make sense to express these with a single op?  It seems that clock-edge driven memories and other stateful things would require fairly different analysis and transformation, thus having them be the same ir node wouldn't make sense.